Created Simple Text Summary application and CD

Lev0nid opened the pull request.

Lev0nid committed:
Implemented application

Implented frontend and backend of the application

Lev0nid committed:
Moved requirenments.txt

Lev0nid committed:
fixed typo in file name

Lev0nid committed:
Added "How to use" section to documentation

Lev0nid committed:
Minor fixes to readme.md

Lev0nid committed:
Small changes to readme.md

Rephrased link name

Lev0nid committed:
Added error message and minor changes

Added error message when input is empty
Changed some displayed texts
Changed output file name generation slightly

Lev0nid committed:
fixed .gitignore and minor fixes to main.py

Lev0nid committed:
Update readme.md

Lev0nid committed:
Update readme.md

Improved build section

Lev0nid committed:
Update readme.md

SpirinEgor commented on backend/process.py:
```diff
@@ -0,0 +1,12 @@
+import torch
+from transformers import pipeline
+
+summarizer = pipeline(
```
As in other languages, making something global isn't a good solution.

Lev0nid replied:
Fixed

SpirinEgor commented on backend/process.py:
```diff
@@ -0,0 +1,12 @@
+summarizer = pipeline(
+    "summarization",
+    "pszemraj/long-t5-tglobal-base-16384-book-summary",
+    device=0 if torch.cuda.is_available() else -1,
```
Conventionally, it is preferred to use "cuda" and "cpu" strings for devices
```python
"cuda" if torch.cuda.is_available() else "cpu"
```

Lev0nid replied:
Fixed

SpirinEgor commented on main.py:
```diff
@@ -1,3 +1,26 @@
+from backend import process
 
-st.title('Work in progress')
+st.title('Simple Text Summary')
```
Don't make it global

Lev0nid replied:
Fixed

SpirinEgor reviewed (changes requested):
Nice 👍 
There are some comments that you need to take care of.

Also, there is no CD setup in this PR. I'm expecting either setup in this PR or separately.

Lev0nid commented:
> Also, there is no CD setup in this PR. I'm expecting either setup in this PR or separately.

I've set up streamlit cloud. For instance, whenever this pull request is merged, the release version of the application will be updated automatically. The test version of the application is updated with each commit to the development branch.

Lev0nid committed:
Resolved issues
Created a class for model, main is now in a function, edited device choosing line

SpirinEgor commented:
I don't see any settings for CD, therefore I can't validate it. Moreover, the link from Readme goes to failed app 🤷‍♂️
